### PR TITLE
[dev-v2.7] adding paths filtering code to validation-check

### PR DESCRIPTION
**Pull request by nicholasSUSE**
Propagation from: https://github.com/rancher/charts/pull/3514

## Problem
Our previous solution created another problem for the `validation-check workflow`. 
Sometimes the validation was required but the previous `validation-comment workflow` was not run because of the filters.

## Solution
Add the same filters for (`assets/ packages/ charts/`) directories to `validation-check workflow`. 
This workflow uses a different type of `github event` so the `paths` directive was not enabled for use. 
The solution implements the same functionality with `javascript` injected code in the workflow. 

## Updated Scope of PR
As discussed below we updated:
- actions/github-script@v4 -> actions/github-script@v7
- actions/checkout@v3 -> actions/checkout@v4

The main reason:  Warnings about the deprecation of old Node.js versions being used since Github Actions is now using an updated version. 

### Manual Testing

All changes were manually tested by:
1. On Forked Repository merging a branch with the changes
2. Creating a new `release-v-test` branch from the recently created one. 
3. Creating a Pull Request with changes (branch: `cicd-test`) that will trigger the workflows and ensure it is working. 

One of the closed Pull Requests in the forked repository used for testing: https://github.com/nicholasSUSE/charts/pull/14

#### Some needed changes after the update:
- https://github.com/marketplace/actions/github-script
- https://octokit.github.io/rest.js/v20

Javascript code like the following example: 
```
            const files = await github.pulls.listFiles({
```

Was updated to:
```
            const files = await github.rest.pulls.listFiles({
```

Because of changes in the `octokit` stated above in the bullet points.

### CODE REVIEW

File: .github/workflows/validation-check.yaml [hunk 1]
```diff
@@ -10,7 +10,26 @@ jobs:
     if: startsWith(github.event.pull_request.base.ref, 'dev-v') || startsWith(github.event.pull_request.base.ref, 'release-v')
     runs-on: ubuntu-latest
     steps:
+      - name: Check for modifications in specified folders
+        id: check-modifications
+        uses: actions/github-script@v4
```
Comment (recena) [inline .github/workflows/validation-check.yaml]: @nicholasSUSE Is there any reason to use an [old version](https://github.com/marketplace/actions/github-script) of this action? In fact, this version is raising some warnings in the log messages.

Comment (nicholasSUSE) [reply .github/workflows/validation-check.yaml]: @recena 
Forgive me I did not see that before. 
Looking on Warnings, other software need updates:
```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3.
```

I will update them all on these PRs.

File: .github/workflows/pull-request.yaml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout base branch
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```
Comment (recena) [inline .github/workflows/pull-request.yaml:15]: You should note these changes in the PR description.

File: .github/workflows/validation-comment.yaml [hunk 3]
```diff
@@ -17,11 +17,11 @@ jobs:
     permissions: write-all
     steps:
       - name: Make validation comment
-        uses: actions/github-script@v4
+        uses: actions/github-script@v7
         with:
           github-token: ${{secrets.GITHUB_TOKEN}}
           script: |
-            github.issues.createComment({
+            github.rest.issues.createComment({
```
Comment (recena) [inline .github/workflows/validation-comment.yaml:24]: How was this tested before?

Comment (nicholasSUSE) [reply .github/workflows/validation-comment.yaml:24]: I am still testing.

Comment (nicholasSUSE) [reply .github/workflows/validation-comment.yaml:24]: I am manually testing all workflows.
That is how I saw that this needed to be changed in the version 7

File: .github/workflows/pull-request.yaml (hunk 2 above)
Comment (nicholasSUSE) [reply .github/workflows/pull-request.yaml:15]: They will be listed once the PR is ready. 
I am not able to convert it again to a draft PR.
Once it is ready I will communicate on slack and jira.

**Comment by recena** [comment]
@nicholasSUSE This branch is very active. Before merging, I will try these changes in `dev-v2.9`

👍 LGTM

**Comment by nicholasSUSE** [comment]
> @nicholasSUSE This branch is very active. Before merging, I will try these changes in `dev-v2.9`
> 
> 👍 LGTM

Currently, we don't have a `dev-v2.9` branch. 
We have a `release-v2.9` that has the modifications of the first commit of this PR but not the ones from the 2nd commit. 

But I agree with your approach, I think it is good for us to implement these changes on `release-v2.9` first and see how it goes.

**Comment by recena** [comment]
@nicholasSUSE 

> Currently, we don't have a dev-v2.9 branch.

`dev-v2.9` is the [default branch](https://github.com/rancher/charts/branches).

**Comment by nicholasSUSE** [comment]
> @nicholasSUSE
> 
> > Currently, we don't have a dev-v2.9 branch.
> 
> `dev-v2.9` is the [default branch](https://github.com/rancher/charts/branches).

You are right. 
Sorry I got my information wrong.

**Review by adamkpickering — APPROVED** [review]

**Comment by nicholasSUSE** [comment]
This change is being implemented in the sync up: https://github.com/rancher/charts/pull/3559

Closing this PR

**Pull request closed without merging**